File: backend/app/schemas/responses/organization.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class OrganizationResponse(BaseModel):
    """How an organization looks when returned to the client."""

    id: UUID
    name: str
    slug: str
    is_active: bool
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class OnBoardingResponse(BaseModel):
    """Schema for the successful onboarding response."""

    organization_id: UUID
    user_id: UUID

File: nidus-frontend/src/modules/organization/types/organization.ts
import { OrganizationMember } from "./members";

export interface OrganizationResponse {
  id: string;
  name: string;
  slug: string;
  is_active: boolean;
  created_at: string;
  members: OrganizationMember[];
}

export interface OnboardingResponse {
  organization_id: string;
  user_id: string;
}

export interface OrganizationCreate {
  organization_name: string;
  organization_slug: string;
  admin_email: string;
  password: string;
}
